### PR TITLE
 8339836: Open source several AWT Mouse tests - Batch 1

### PR DESCRIPTION
**Pull request by TejeshR13**
Open source these AWT Mouse tests:

java/awt/Mouse/MouseEnterExitTest.java
java/awt/Mouse/MouseEnterExitTest2.java
java/awt/Mouse/MousePressedTest.java
java/awt/Mouse/MouseEnterExitTest4.java
java/awt/Mouse/MouseEnterExitTest3.java

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8339836](https://bugs.openjdk.org/browse/JDK-8339836): Open source several AWT Mouse tests - Batch 1 (**Bug** - P4)

### Reviewers
 * [Harshitha Onkar](https://openjdk.org/census#honkar) (@honkar-jdk - **Reviewer**)
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/21124/head:pull/21124` \
`$ git checkout pull/21124`

Update a local copy of the PR: \
`$ git checkout pull/21124` \
`$ git pull https://git.openjdk.org/jdk.git pull/21124/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 21124`

View PR using the GUI difftool: \
`$ git pr show -t 21124`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/21124.diff">https://git.openjdk.org/jdk/pull/21124.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/21124#issuecomment-2367244663)

### CODE REVIEW

File: test/jdk/java/awt/Mouse/MouseEnterExitTest.java [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+import java.awt.event.WindowAdapter;
+import java.awt.event.WindowEvent;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 4050138
+ * @summary Test to verify Lightweight components don't get
+ *          enter/exit during drags
+ * @library /java/awt/regtesthelpers
```
Comment (prrace) [inline test/jdk/java/awt/Mouse/MouseEnterExitTest.java:46]: Can't the drag be done by Robot to make this an automated test ?

Comment (TejeshR13) [reply test/jdk/java/awt/Mouse/MouseEnterExitTest.java:46]: Yes, converted to automated test. Tested in CI too.

**Review by honkar-jdk — APPROVED** [review]
LGTM other than few inline suggestions

File: test/jdk/java/awt/Mouse/MouseEnterExitTest4.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import java.awt.event.MouseAdapter;
+import java.awt.event.MouseEvent;
+import java.awt.event.MouseListener;
+import java.util.List;
+
+/*
+ * @test
+ * @bug 4431868
+ * @summary Tests that window totally obscured by its child doesn't receive
+ *          enter/exit events when located over another frame
```
Comment (honkar-jdk) [inline test/jdk/java/awt/Mouse/MouseEnterExitTest4.java:37]: Suggestion: This test can be automated similar to MouseEnterExitTest.java  with robot.mouseMove() and checking for mouseEntered and Exited events.

**Review by prrace — APPROVED** [review]

**Review by honkar-jdk — APPROVED** [review]
Updated changes to make MouseEnterExitTest4.java automated looks good.

**Pull request closed without merging**